DigitalLearningSolutions.Data/Services/SectionContentService.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System.Data;
+    using System.Linq;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Models.SectionContent;
+    using Microsoft.Extensions.Logging;
+
+    public interface ISectionContentService
+    {
+        SectionContent? GetSectionContent(int customisationId, int progressId, int sectionId);
+    }
+
+    public class SectionContentService : ISectionContentService
+    {
+        private readonly IDbConnection connection;
+        private readonly ILogger<SectionContentService> logger;
+
+        public SectionContentService(IDbConnection connection, ILogger<SectionContentService> logger)
+        {
+            this.connection = connection;
+            this.logger = logger;
+        }
+
+        public SectionContent? GetSectionContent(int customisationId, int progressId, int sectionId)
+        {
+            return connection.Query<SectionContent>(
+                @"SELECT Sections.SectionName, 
```

Comment:
The indentation is a bit inconsistent in this query. Fixing that will make it a bit easier to read 🙂

Reply:
I wasn't really sure of the best way to indent the "FROM" part of the query as there is a sort of nested JOIN. I chose to indent the nested part as follows
```
FROM aspProgress
INNER JOIN Progress ON aspProgress.ProgressID = Progress.ProgressID 
INNER JOIN Sections 
    INNER JOIN Tutorials ON Sections.SectionID = Tutorials.SectionID 
    INNER JOIN CustomisationTutorials ON Tutorials.TutorialID = CustomisationTutorials.TutorialID 
ON aspProgress.TutorialID = Tutorials.TutorialID                    
```
Is there a better way to do this?

Reply:
Good point, it is a bit of a tricky one. I think just a few tweaks are necessary to make sure all statements at the same level are indented the same. So everything in the SELECT section should be indented the same, the inner joins at the same level should be indented the same etc. Does that make sense? For the from section I'd probably do something like 
```
FROM aspProgress
  INNER JOIN Progress ON aspProgress.ProgressID = Progress.ProgressID 
  INNER JOIN Sections 
    INNER JOIN Tutorials ON Sections.SectionID = Tutorials.SectionID 
    INNER JOIN CustomisationTutorials ON Tutorials.TutorialID = CustomisationTutorials.TutorialID 
    ON aspProgress.TutorialID = Tutorials.TutorialID 
  INNER JOIN Customisations ON Progress.CustomisationID = Customisations.CustomisationID
  LEFT OUTER JOIN AssessAttempts ON Progress.ProgressID = AssessAttempts.ProgressID AND Sections.SectionNumber = AssessAttempts.SectionNumber
WHERE
```
🙂